### PR TITLE
test(server): Fix flaky shutdown test

### PR DESCRIPTION
**Pull request by jjbayer**
Simplify test of `drop_guard`.

#skip-changelog

### CODE REVIEW

File: relay-server/src/utils/shutdown.rs [hunk 1]
```diff
@@ -70,24 +70,34 @@ pub trait FutureExt: Sized {
...
-                .then(|_| Ok(())),
-        );
-    });
+            actix::spawn(
+                relay_test::delay(Duration::from_millis(2000))
+                    .drop_guard("bye")
+                    .then(|_| Ok(())),
+            );
+
+            test_now.advance(Duration::from_millis(1500));
```
Comment (jan-auer) [inline relay-server/src/utils/shutdown.rs]: Now that we're driving the clock manually, can we just create the future, advance a bit, then stop the system without the second delay?

File: relay-test/src/lib.rs [hunk 2]
```diff
@@ -121,5 +122,55 @@ where
...
+
+impl TestNow {
+    pub fn new() -> Self {
+        TestNow {
+            current: Arc::new(RwLock::new(Instant::now())),
+        }
+    }
+
+    pub fn advance(&mut self, duration: Duration) {
+        *self.current.write().unwrap() += duration;
```
Comment (jan-auer) [inline relay-test/src/lib.rs]: How do you tell the actor system or executor that a tick has occurred?

Comment (jjbayer) [reply relay-test/src/lib.rs]: You pass the clock to the `System` builder:

```
System::builder().clock(test_clock);
```

Comment (jan-auer) [reply relay-test/src/lib.rs]: That makes sense, but what polls the clock. When you call advance, the executor is waiting on delay futures, so how does it get notified to poll the clock again? Usually, you would have to call a waker.

File: relay-server/src/utils/shutdown.rs (hunk 1 above)
Comment (jjbayer) [reply relay-server/src/utils/shutdown.rs]: Or reduce the test to this, in which case we don't even need the manually driven clock?

```rust
System::run(move || {
    actix::spawn(
        relay_test::delay(Duration::from_secs(10))
            .drop_guard("bye")
            .then(|_| Ok(())),
    );

    actix::System::current().stop();
});
```

Comment (jan-auer) [reply relay-server/src/utils/shutdown.rs]: Yes, that makes sense. We should then have another test that ensures we don't panic when the future completes, however. Something like:

```rust
System::run(move || {
    actix::spawn(
        relay_test::delay(Duration::from_secs(10))
            .drop_guard("bye")
            .then(|_| {
                actix::System::current().stop();
                Ok(())
            }),
    );
});
```

**Review by jan-auer — APPROVED** [review]

**Pull request merged**